pr2eus/robot-interface.l

```diff
@@ -1592,17 +1592,23 @@ Return value is a list of interpolatingp for all controllers, so (null (some #'i
      (send self :move-to-wait :retry 1 :frame-id base-frame-id :no-wait nil)
      ))
   (:send-cmd-vel-raw
-   (x y d) ;; [m] [m] [degree]
+   (x y d &key topic-name) ;; [m] [m] [degree]
    (when (send self :simulation-modep)
      (return-from :send-cmd-vel-raw t))
-   (unless (ros::get-topic-publisher (if namespace (format nil "~A/~A" namespace cmd-vel-topic) cmd-vel-topic))
-     (ros::advertise (if namespace (format nil "~A/~A" namespace cmd-vel-topic) cmd-vel-topic) geometry_msgs::Twist 1)
+   (let ((vel-topic
+         (if topic-name
+             (if namespace
+                 (if (eq (char "/" 0) (char topic-name 0)) topic-name (format nil "~A/~A" namespace topic-name))
+                 topic-name)
+             (if namespace (format nil "~A/~A" namespace cmd-vel-topic) cmd-vel-topic))))
```

Comment:
```lisp
  (:send-cmd-vel-raw
   (x y d &key (topic-name cmd-vel-topic) ((:namespace ns) namespace)) ;; [m] [m] [degree]
    (if ns (setq topic-name (format nil "~A/~A" ns topic-name)))
    ;; ...
```

Reply:
Thank you for the helpful advice. It's simple and easy to understand.